DigitalLearningSolutions.Web/ServiceFilter/VerifyDelegateCanAccessActionPlanItem.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace DigitalLearningSolutions.Web.ServiceFilter
+{
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Helpers;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Filters;
+
+    public class VerifyDelegateCanAccessActionPlanItem : IActionFilter

```

Comment:
I created a similar service filter in 664 but this one is much more comprehensive. I'll remove mine from my MR and then whoever's ticket is merged second can add this attribute to the controller methods added in 664 :)